modules/appeals_api/app/workers/appeals_api/add_icn_updater.rb

```diff
@@ -35,5 +45,32 @@ def target_veteran(appeal)
 
       mpi_veteran
     end
+
+    def target_veteran_with_address(appeal)
+      veteran ||= Appellant.new(
+        type: :veteran,
+        auth_headers: appeal.auth_headers,
+        form_data: appeal.form_data&.dig('data', 'attributes', 'veteran')
+      )
+
+      # AppealsApi::Veteran is a wrapper for ClaimsApi::Veteran, which requires ssn.
+      # Instead of modifying ClaimsApi::Veteran, we'll define the following missing methods
+      # we need on this veteran object and just call the MPI service directly.
+
+      veteran.define_singleton_method(:birth_date) do
+        appeal.auth_headers['X-VA-Birth-Date']
+      end
+      veteran.define_singleton_method(:valid?) do
+        true
+      end
+      veteran.define_singleton_method(:authn_context) do
+        'authn'
+      end
+      veteran.define_singleton_method(:uuid) do
+        nil
+      end
+
+      MPI::Service.new.find_profile(veteran)
+    end
```

Comment:
this and the `.try` modifications in `lib/mpi/service.rb` feel a little too brittle as currently constructed. Would the best way to adapt this to be defining some kind of wrapper class that's just a wrapper around `AppealsApi::Appellant` and these additional methods needed to appease various MPI checks?

Reply:
>  defining some kind of wrapper class...

Did that [here](https://github.com/department-of-veterans-affairs/vets-api/pull/10574/commits/97472f0600f7cd582f690096b2a9d06abfc52ffa) and [here](https://github.com/department-of-veterans-affairs/vets-api/pull/10574/commits/3cc41e16d00f0c3261fa8eaaaff3255b03afede5).